dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CheckUrlHelper.java

```diff
@@ -69,112 +63,47 @@ private static Optional<Boolean> checkUrl(String url, String baseURL) {
             Thread.currentThread().interrupt();
             return Optional.empty();
         }
     }
 
-    private static Optional<Boolean> checkUrls(Set<String> urls, String baseURL) {
-        List<Optional<Boolean>> objectStream = urls.parallelStream().map(url -> checkUrl(url, baseURL))
-            .collect(Collectors.toCollection(ArrayList::new));
-        if (objectStream.stream().anyMatch(urlStatus -> urlStatus.isPresent() && urlStatus.get().equals(false))) {
-            return Optional.of(false);
-        }
-        if (objectStream.stream().anyMatch(Optional::isEmpty)) {
-            return Optional.empty();
-        }
-        return Optional.of(true);
+    private static boolean hasMalformedUrl(Set<String> possibleUrls) {
+        return possibleUrls.stream().anyMatch(possibleUrl -> {
+            try {
+                new URL(possibleUrl);
+            } catch (MalformedURLException e) {
+                LOGGER.debug("malformed url", e);
+                return true;
+            }
+            return false;
+        });
     }
 
     /**
-     * Get all the URLs from a JSON file.
-     *
-     * @param contents Contents of a JSON file
-     * @return The URLs
+     * Checks whether all <code>possibleUrls</code> are open access urls. Returns <code>Optional.of(true)</code>
+     * if they are, <code>Optional.of(false)</code> if any are not, or <code>Optional.empty()</code>
+     * if there was an error doing the check.
+     * @param possibleUrls
+     * @param baseURL
+     * @return
      */
-    public static Set<String> getUrlsFromJSON(String contents) {
-        Set<String> urls = new HashSet<>();
-        JSONObject jsonFile = new JSONObject(contents);
-        getUrlsFromJSON(jsonFile, urls);
-        return urls;
-    }
-
-    private static void getUrlsFromJSON(Object object, Set<String> urls) {
-        if (object instanceof JSONObject) {
-            getUrlsFromJSON((JSONObject) object, urls);
-        }
-        if (object instanceof String) {
-            getUrl((String) object, urls);
-        }
-        if (object instanceof JSONArray) {
-            getUrlsFromJSON((JSONArray) object, urls);
-        }
-        // Ignore instanceof boolean, number, or null
-    }
-
-    private static void getUrlsFromJSON(JSONArray jsonArray, Set<String> urls) {
-        for (int i = 0; i < jsonArray.length(); i++) {
-            getUrlsFromJSON(jsonArray.get(i), urls);
-        }
-    }
-
-    private static void getUrlsFromJSON(JSONObject jsonObject, Set<String> urls) {
-        Iterator<?> keys = jsonObject.keys();
-        while (keys.hasNext()) {
-            String key = (String) keys.next();
-            Object o = jsonObject.get(key);
-            getUrlsFromJSON(o, urls);
+    public static Optional<Boolean> checkUrls(Set<String> possibleUrls, String baseURL) {
+        if (possibleUrls.isEmpty() || hasMalformedUrl(possibleUrls)) {
+            // If there are no urls, or any values that aren't urls, it's not open data
```

Comment:
> Seems intentional, so asking the question. Wouldn't "no urls" be essentially open data? i.e. a workflow that runs with no parameters or requires no input

You could argue this both ways, and I could go either way. Should a "Hello world" workflow be considered open data? For now, I opted that it's not, that no data != open data.  On the other hand, if you're looking for workflows you can run without restricted access data, it would be useful.

Maybe we should come up with another term instead of open data, something that indicates runnable as is without any access controlled data. Trying to think of one.

Reply:
Ok with a third term, without a third term I'd argue that yes they should be open data. 
Imagine, yes, a hello world workflow, a workflow that only takes integer parameters, or something like a chatgpt container that only takes a String question

Reply:
Brainstorming here, an idea would be to reframe the terminology, and use the term "open" instead of "open data".  A workflow is "open" if it doesn't require restricted resources...

Reply:
I don't think we can detect if a workflow requires restricted resources. Well, not today -- we'd have to scan the code and see if there are any hard-coded references to restricted resources. We can only detect if a test parameter file references open/restricted resources. Here are the scenarios:

Id | Use Case  | Notes  |
-- | ---------- | -------- |
1 | Workflow has no input parameters  |     |
2 | Workflow has input parameters, but none are files |  |
3 | Workflow has file input parameters and provides a test parameter file with open access data for every file input| |
4 | Workflow has file input parameters and has no test param file, or has no test param file without open access data for every file parameter | Note that many test param files have references to local data, i.e., relative file paths, that could be open data, but we have no way of knowing |

The PR currently only flags option 3 as open access. Denis wants cases 1 and 2 covered as well.

So, we want to be able to find workflows that, as registered on Dockstore, are completely runnable on their own as they are on Dockstore (except for Docker and the workflow engine), without any additional access required.

Brainstorm of ideas, none of which I like. I'm hoping they may trigger a good idea for somebody :). Keep in mind we'll need to surface this in the search facets.

* Runnable-as-is; Runnable locally; Runnable
* Self-contained
* Complete; Complete bundle

I'll try to think of something better over the weekend. :)

Reply:
> The PR currently only flags option 3 as open access. Denis wants cases 1 and 2 covered as well.

:heavy_check_mark: 

I think we just use "open data" vs "closed data" and just have a :information_source:  icon or something explaining. I think people are kinda used to the zero or null case being a little weird.